civis/ml/_model.py

```diff
@@ -450,3 +462,518 @@ def training_metadata(self):
+            Name of the database holding the training set table used to
+            build the model. E.g., 'My Cluster Name'.
+        file_id : int, optional
+            If the training data are stored in a Civis file,
+            provide the integer file ID.
+        sql_where : str, optional
+            A SQL WHERE clause used to scope the rows of the training set
+            (used for table input only)
+        sql_limit : int, optional
+            SQL LIMIT clause for querying the training set
+            (used for table input only)
+        oos_scores : str, optional
+            If provided, store out-of-sample predictions on
+            training set data to this Redshift "schema.tablename".
+        oos_scores_db : str, optional
+            If not provided, store OOS predictions in the same database
+            which holds the training data.
+        if_exists : {'fail', 'append', 'drop', 'truncate'}
+            Action to take if the out-of-sample prediction table
+            already exists.
+        fit_params: Dict[str, str]
+            Mapping from parameter names in the model's `fit` method
+            to the column names which hold the data, e.g.
+            `{'sample_weight': 'survey_weight_column'}`.
+        polling_interval : float, optional
+            Check for job completion every this number of seconds.
+            Do not set if using the notifications endpoint.
+
+        Returns
+        -------
+        :class:`~ModelFuture`
+        """
+        if ((table_name is None or database_name is None) and
+                file_id is None and
+                X is None):
+            raise ValueError('Provide a source of data.')
+        if sum((bool(table_name and database_name),
+                bool(file_id),
+                X is not None)) > 1:
+            raise ValueError('Provide a single source of data.')
+
+        if X is not None:
+            file_id = _stash_local_data(X, client=self._client)
+
+        train_args = {'TARGET_COLUMN': ' '.join(self.dependent_variable),
+                      'PRIMARY_KEY': self.primary_key,
+                      'PARAMS': json.dumps(self.parameters),
+                      'CVPARAMS': json.dumps(self.cv_params),
+                      'CALIBRATION': self.calibration,
+                      'IF_EXISTS': if_exists}
+        if oos_scores:
+            train_args['OOSTABLE'] = oos_scores
+        if oos_scores_db:
+            oos_db_id = self._client.get_database_id(oos_scores_db)
+            train_args['OOSDB'] = {'database': oos_db_id}
+        if sql_where:
+            train_args['WHERESQL'] = sql_where
+        if sql_limit:
+            train_args['LIMITSQL'] = sql_limit
+        if self.excluded_columns:
+            train_args['EXCLUDE_COLS'] = ' '.join(self.excluded_columns)
+        if fit_params:
+            train_args['FIT_PARAMS'] = json.dumps(fit_params)
+
+        if HAS_SKLEARN and isinstance(self.model, BaseEstimator):
+            import joblib
```

Comment:
Why put the `import` here rather than in the `try` above where sklearn is imported?